include/zephyr/dt-bindings/gpio/nxp-imx-gpio.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2025 Feniex Industries Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_NXP_IMX_GPIO_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_NXP_IMX_GPIO_H_
+
+/**
+ * @name GPIO pull strength flags
+ *
+ * The pull strength flags are a Zephyr specific extension of the standard GPIO
+ * flags specified by the Linux GPIO binding. Only applicable for NXP IMX
+ * SoCs.
+ *
+ * The interface supports two different pull strengths:
+ * `WEAK` - The lowest pull strength supported by the HW
+ * `STRONG` - The highest pull strength supported by the HW
+ *
+ * @{
+ */
+/** @cond INTERNAL_HIDDEN */
+#define IMXRT10XX_GPIO_PULL_STRENGTH_POS 8
```

Comment:
Naming suggestion. I would not make this specific to RT10xx. Instead can we rename these to prefix with `NXP_IGPIO_`
e,g: `NXP_IGPIO_PULL_STRENGTH_POS` etc.